### ISSUE TITLE
v1.7 does not build on skypack

### ISSUE THREAD

**Issue by xunilrj** [solidjs/solid#1660]
### Describe the bug

skypack fails to build v1.7: https://cdn.skypack.dev/solid-js/web

```
/*
 * [Package Error] "solid-js@v1.7.0" could not be built. 
 *
 *   [1/5] Verifying package is valid…
 *   [2/5] Installing dependencies from npm…
 *   [3/5] Building package using esinstall…
 *   Running esinstall...
 *   rollup-plugin-inject: failed to parse solid-js/dist/dev.cjs. Consider restricting the plugin to particular files via options.include
 *   /tmp/cdn/_VNezyagvbSfhA4EpFC9I/node_modules/solid-js/dist/dev.cjs:1794:18 Error when using sourcemap for reporting an error: Can't resolve original location of error.
 *   Failed to load node_modules/solid-js/dist/dev.cjs
 *     Duplicate export 'DEV' (Note that you need plugins to import files that are not JavaScript)
 *   Install failed.
 *   Install failed.
 *
 * How to fix:
 *   - If you believe this to be an error in Skypack, file an issue here: https://github.com/skypackjs/skypack-cdn/issues
 *   - If you believe this to be an issue in the package, share this URL with the package authors to help them debug & fix.
 *   - Use https://skypack.dev/ to find a web-friendly alternative to find another package.
 */

console.warn("[Package Error] \"solid-js@v1.7.0\" could not be built. \n[1/5] Verifying package is valid…\n[2/5] Installing dependencies from npm…\n[3/5] Building package using esinstall…\nRunning esinstall...\nrollup-plugin-inject: failed to parse solid-js/dist/dev.cjs. Consider restricting the plugin to particular files via options.include\n/tmp/cdn/_VNezyagvbSfhA4EpFC9I/node_modules/solid-js/dist/dev.cjs:1794:18 Error when using sourcemap for reporting an error: Can't resolve original location of error.\nFailed to load node_modules/solid-js/dist/dev.cjs\n  Duplicate export 'DEV' (Note that you need plugins to import files that are not JavaScript)\nInstall failed.\nInstall failed.");
throw new Error("[Package Error] \"solid-js@v1.7.0\" could not be built. ");
export default null;
```

### Your Example Website or App

 https://cdn.skypack.dev/solid-js/web

### Steps to Reproduce the Bug or Issue

1. Just navigate to https://cdn.skypack.dev/solid-js/web

### Expected behavior

v1.7 should build successfully.

### Platform

Not sure what skypack uses. But probably linux.

### Additional context

Not sure it is the same problem, but locally, it seems it is missing an ```"workspaces": [...]``` in the packages.json.

**Comment by Tete1805** [comment]
This seems to impact the website's playground as well: https://www.solidjs.com/tutorial/introduction_basics

In the playground's console:
```
The requested module 'https://cdn.skypack.dev/solid-js/web' does not provide an export named 'createComponent'
```

When you look at the request file at https://cdn.skypack.dev/error/build:solid-js@v1.7.0-VNezyagvbSfhA4EpFC9I, the content reads as follows:
```
/*
 * [Package Error] "solid-js@v1.7.0" could not be built. 
 *
 *   [1/5] Verifying package is valid…
 *   [2/5] Installing dependencies from npm…
 *   [3/5] Building package using esinstall…
 *   Running esinstall...
 *   rollup-plugin-inject: failed to parse solid-js/dist/dev.cjs. Consider restricting the plugin to particular files via options.include
 *   /tmp/cdn/_VNezyagvbSfhA4EpFC9I/node_modules/solid-js/dist/dev.cjs:1794:18 Error when using sourcemap for reporting an error: Can't resolve original location of error.
 *   Failed to load node_modules/solid-js/dist/dev.cjs
 *     Duplicate export 'DEV' (Note that you need plugins to import files that are not JavaScript)
 *   Install failed.
 *   Install failed.
 *
 * How to fix:
 *   - If you believe this to be an error in Skypack, file an issue here: https://github.com/skypackjs/skypack-cdn/issues
 *   - If you believe this to be an issue in the package, share this URL with the package authors to help them debug & fix.
 *   - Use https://skypack.dev/ to find a web-friendly alternative to find another package.
 */

console.warn("[Package Error] \"solid-js@v1.7.0\" could not be built. \n[1/5] Verifying package is valid…\n[2/5] Installing dependencies from npm…\n[3/5] Building package using esinstall…\nRunning esinstall...\nrollup-plugin-inject: failed to parse solid-js/dist/dev.cjs. Consider restricting the plugin to particular files via options.include\n/tmp/cdn/_VNezyagvbSfhA4EpFC9I/node_modules/solid-js/dist/dev.cjs:1794:18 Error when using sourcemap for reporting an error: Can't resolve original location of error.\nFailed to load node_modules/solid-js/dist/dev.cjs\n  Duplicate export 'DEV' (Note that you need plugins to import files that are not JavaScript)\nInstall failed.\nInstall failed.");
throw new Error("[Package Error] \"solid-js@v1.7.0\" could not be built. ");
export default null;
```

**Comment by ryansolid** [comment]
Interesting. I didn't think we changed anything module specific here that would cause this. I wonder which version introduced it. That's probably where I would start my search. Starting with 1.6.16.

**Comment by ryansolid** [comment]
Ok looks like this was introduced right from the first build 1.7.0-beta.0. Funny enough I have no idea what the issue is. The generated Solid code looks fine. This has to be some name collision with some symbol or something it is complaining about multiple declarations of DEV. But there is only one.

**Comment by xunilrj** [comment]
Well `packages/solid/dist/dev.cjs` contains the below. Maybe this is what skypack is complaining:

```js
exports.DEV = void 0;
exports.DEV = {
  hooks: DevHooks,
  writeSignal,
  registerGraph
};
```

It seems an optimizer bug... it is wrongly exporting DEV twice from 

```js
let DEV:
  | {
      readonly hooks: typeof DevHooks;
      readonly writeSignal: typeof writeSignal;
      readonly registerGraph: typeof registerGraph;
    }
  | undefined;
if ("_SOLID_DEV_") DEV = { hooks: DevHooks, writeSignal, registerGraph };
export { DEV };
```

Especially because "_SOLID_DEV_" is being replaces by `true`
One ugly solution is to make it harder to optimize like this: 

```js
const DEV = (() => {
  if ("_SOLID_DEV_")
    return { hooks: DevHooks, writeSignal, registerGraph } as {
      readonly hooks: typeof DevHooks;
      readonly writeSignal: typeof writeSignal;
      readonly registerGraph: typeof registerGraph;
    };
  else return undefined;
})();
export { DEV };
```

I can create a PR to give it a try.

**Comment by thetarnav** [comment]
I think that 1.6 had a similar reassigning. 1.7 only changed the contents of the object and the type.
Could you check if v1.6 has the same duplicated `exports.DEV = ...` line?

**Comment by ryansolid** [comment]
Fixed initial error and now it fails without any indication why.

**Comment by xunilrj** [comment]
Maybe a skypack issue now? Seeing other packages with the same error. :(

https://github.com/skypackjs/skypack-cdn/issues/345
https://github.com/skypackjs/skypack-cdn/issues/344
https://github.com/skypackjs/skypack-cdn/issues/339

**Comment by kidonng** [comment]
Skypack seems to be lacking maintenance since a long time ago, since the maintainer has shifted focus to Astro. I have had various issues using Skypack in Deno.

I would suggest moving to esm.sh.

**Comment by kidonng** [comment]
It seems solid-repl had [switched to esm.sh](https://github.com/solidjs/solid-playground/pull/119) 8 months ago, and recently [switched to jspm.dev](https://github.com/solidjs/solid-playground/pull/136), it's just not released.

@modderme123 would you like to do a release and upte solid-site to resolve the issue?

**Comment by modderme123** [comment]
We're working on it, there are a number of rollup bugs that have popped up in recent versions of packages that are making it more difficult

**Comment by ryansolid** [comment]
Ok skypack is fixed in 1.7.2. Just needed to go back to exporting as a single const with the ternary.

So weird but we seem to be good now.
